tests/framework/e2e/etcd_process.go

```diff
@@ -201,11 +202,35 @@ func (ep *EtcdServerProcess) Kill() error {
 	return ep.proc.Signal(syscall.SIGKILL)
 }
 
-func (ep *EtcdServerProcess) Wait() error {
-	ep.proc.Wait()
+func (ep *EtcdServerProcess) Wait(ctx context.Context) error {
+	ch := make(chan struct{})
+	go func() {
+		defer close(ch)
+		if ep.proc != nil {
+			ep.proc.Wait()
+			ep.cfg.lg.Info("server exited", zap.String("name", ep.cfg.Name))
+		}
+	}()
+	select {
+	case <-ch:
+		ep.proc = nil
+		return nil
+	case <-ctx.Done():
+		return ctx.Err()
```

Comment:
This logic is leaking go-routine if I understands correctly... 
We will exit with error, but we will keep waiting on the ep.proc.Wait(). 

Eventually the process we will finish and the go-routine will get cleaned up. Just be aware that it can lead to goroutine-leak detection (as long as we don't pass ep.proc.Wait(ctx) that enables Wait cancelation when context is done.